backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/KotlinObjCClassInfoGenerator.kt

```diff
@@ -27,10 +27,17 @@ internal class KotlinObjCClassInfoGenerator(override val context: Context) : Con
 
         val instanceMethods = generateInstanceMethodDescs(irClass)
 
-        val companionObject = irClass.declarations.filterIsInstance<IrClass>().atMostOne { it.isCompanion  }
+        val companionObject = irClass.companionObject()
         val classMethods = companionObject?.generateMethodDescs().orEmpty()
 
-        val superclassName = irClass.getSuperClassNotAny()!!.let {
+        val superClass = irClass.getSuperClassNotAny()!!
+        if (superClass.isObjCClass() && companionObject != null) {
+            val companionBodySize = LLVMStoreSizeOfType(llvmTargetData, context.llvmDeclarations.forClass(companionObject).bodyType).toInt()
```

Comment:
It would be more straightforward and reliable to check presence of `IrField`s in the class instead.

Reply:
I tried both approaches, they have the same effect of course. My choice was based on the same "straightforward" rationale: I want to check for non-trivial body which we don't initialize correctly, and size (i.e. non-initialized memory slots) is exactly what differentiate it from trivial one. Also, checking the size is simpler, it works for backing fields, etc. Saying that, I would agree that testing against pointer size is not exactly "trivial body test" and may be fragile. So I have no clear preferences.

Reply:
Thank you for the explanation.
We have plans to migrate to frontend all these compilation errors reported by backend. So adding this check to InteropLowering will simplify the migration for the check.

Checking this on IR level instead of "body-size" level makes it more straightforward to report location of particular field instead of class (see https://github.com/JetBrains/kotlin-native/pull/4054#discussion_r402954213).